dupedetection/client.go

```diff
@@ -0,0 +1,127 @@
+package dupedetection
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pastelnetwork/gonode/common/errors"
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/pastelnetwork/gonode/common/random"
+)
+
+type client struct {
+	config *Config
+}
+
+// Generate implements dupedection.Client.Generate
+func (client *client) Generate(ctx context.Context, path string) (*DupeDetection, error) {
+
+	// Copy image file to dupe detection service input directory
+	outputBase, outputPath, err := client.copyImageToInputDir(path)
+	if err != nil {
+		return nil, errors.Errorf("failed to copy image to dupe detection input directory: %w", err)
+	}
+
+	// Collect result from dupe detection output directory
+	result, err := client.collectOutput(ctx, outputBase, outputPath)
+	if err != nil {
+		err = errors.Errorf("failed to collect output from dupe detection service: %w", err)
+	}
+
+	return result, err
+}
+
+func (client *client) copyImageToInputDir(inputPath string) (string, string, error) {
+	input, err := ioutil.ReadFile(inputPath)
+	if err != nil {
+		return "", "", err
+	}
+
+	ext := filepath.Ext(inputPath)
+
+	// Generate random name
+	inputName, err := random.String(10, random.Base62Chars)
+	if err != nil {
+		return "", "", err
+	}
+
+	inputPath = filepath.Join(client.config.InputDir, inputName+ext)
+
+	err = ioutil.WriteFile(inputPath, input, os.ModePerm)
+	if err != nil {
+		return "", "", err
+	}
+
+	outputBase := inputName + ".json"
+	outputPath := filepath.Join(client.config.OutputDir, outputBase)
+	return outputBase, outputPath, nil
+}
+
+func (client *client) collectOutput(ctx context.Context, baseName, path string) (*DupeDetection, error) {
+	// Waiting for JSON output from dupe detection service
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
Better if we define the duration time in config (it can get from config file or by default)

Reply:
This point is very small, and we will not touch it in the future, so I decided to define it here. Thanks.